controllers/volumereplication_controller.go

```diff
@@ -64,3 +86,32 @@ func (r *VolumeReplicationReconciler) SetupWithManager(mgr ctrl.Manager, cfg *co
 		For(&replicationv1alpha1.VolumeReplication{}).
 		Complete(r)
 }
+
+func (r VolumeReplicationReconciler) getDataSource(instance *replicationv1alpha1.VolumeReplication) (*corev1.PersistentVolumeClaim, *corev1.PersistentVolume, error) {
+	pvcName := instance.Spec.DataSource.Name
+	pvc := &corev1.PersistentVolumeClaim{}
```

Comment:
currently, we are supporting the only PVC but when we start supporting other sources like a snapshot or anything else till will fail. need to decide this based on the Kind in datasource field.

Reply:
added a switch case to support this for now.  We can enhance it to use any other better approach once we add support for other data sources.